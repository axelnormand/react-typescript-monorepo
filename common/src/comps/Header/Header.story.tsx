import { storiesOf } from '@storybook/react';
import React from 'react';
import Header from './Header';

storiesOf('Header', module).add('with text', () => <Header>Hi</Header>);
