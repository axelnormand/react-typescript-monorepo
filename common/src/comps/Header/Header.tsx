import React from 'react';
import styled from 'styled-components';

export interface HeaderProps {
  children: string;
}

const Header: React.SFC<HeaderProps> = ({ children }) => (
  <StyledHeader>{children}</StyledHeader>
);

const StyledHeader = styled.h1`
  font-size: 35px;
  color: #fbc121;
`;

export default Header;
