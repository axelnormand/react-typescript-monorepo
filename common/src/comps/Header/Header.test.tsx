import 'jest-dom/extend-expect';
import 'jest-styled-components';
import React from 'react';
import { render } from 'react-testing-library';
import Header, { HeaderProps } from './Header';

const props: HeaderProps = {
  children: 'Test',
};

test('renders', () => {
  const { container, getByText } = render(<Header {...props} />);
  const textNode = getByText(props.children);
  expect(textNode).toBeVisible();
  expect(container.firstChild).toHaveStyleRule('font-size', '35px');
  expect(container).toMatchSnapshot();
});
