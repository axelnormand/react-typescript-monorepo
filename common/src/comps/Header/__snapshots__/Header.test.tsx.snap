// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  font-size: 35px;
  color: #fbc121;
}

<div>
  <h1
    class="c0"
  >
    Test
  </h1>
</div>
`;
